app/controllers/spree/api/v1/user_sessions_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Spree
+  module Api
+    module V1
+      class UserSessionsController < Devise::SessionsController
+        skip_before_action :verify_authenticity_token
+
+        respond_to :json
+
+        def create
+          authenticate_spree_user!
+
+          if spree_user_signed_in?
+            respond_to do |format|
+              format.html {
+                flash[:success] = Spree.t(:logged_in_succesfully)
+                redirect_back_or_default(after_sign_in_path_for(spree_current_user))
+              }
+              format.js {
+                render json: { user: spree_current_user,
+                               ship_address: spree_current_user.ship_address,
+                               bill_address: spree_current_user.bill_address }.to_json
+              }
+              format.json {
+                render json: { success: true,
```

Comment:
I dont think we need to return success: true, we already have status code, the client should validate that, so with the format.js should be enough